yarn-project/circuit-types/src/keys/key_store.ts

```diff
@@ -32,12 +32,12 @@ export interface KeyStore {
   getAccounts(): Promise<AztecAddress[]>;
 
   /**
-   * Gets the master nullifier public key for a given account.
+   * Gets the master nullifier public key for a given account or master nullifier public key hash.
    * @throws If the account does not exist in the key store.
-   * @param account - The account address for which to retrieve the master nullifier public key.
+   * @param account or master nullifier public key hash - The account address or master nullifier public key hash for which to retrieve the master nullifier public key.
    * @returns The master nullifier public key for the account.
    */
-  getMasterNullifierPublicKey(account: AztecAddress): Promise<PublicKey>;
+  getMasterNullifierPublicKey(args: { account: AztecAddress } | { npkMHash: Fr }): Promise<PublicKey>;
```

Comment:
this looks a little odd to me, would I not need to call it differently? Why not allowing just AztecAddress or the other struct? Not sure why it is inside `{}`